test/state/rlp.hpp

```diff
@@ -0,0 +1,117 @@
+// evmone: Fast Ethereum Virtual Machine implementation
+// Copyright 2021 The evmone Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+#pragma once
+
+#include <intx/intx.hpp>
+#include <cassert>
+#include <string>
+#include <string_view>
+#include <utility>
+#include <vector>
+
+namespace evmone::rlp
+{
+using bytes = std::basic_string<uint8_t>;
+using bytes_view = std::basic_string_view<uint8_t>;
+
+namespace internal
+{
+template <uint8_t ShortBase, uint8_t LongBase>
+inline bytes encode_length(size_t l)
+{
+    static constexpr auto short_cutoff = 55;
+    static_assert(ShortBase + short_cutoff <= 0xff);
+    assert(l <= 0xffffff);
+
+    if (l <= short_cutoff)
+        return {static_cast<uint8_t>(ShortBase + l)};
+    else if (const auto l0 = static_cast<uint8_t>(l); l <= 0xff)
+        return {LongBase + 1, l0};
+    else if (const auto l1 = static_cast<uint8_t>(l >> 8); l <= 0xffff)
+        return {LongBase + 2, l1, l0};
+    else
+        return {LongBase + 3, static_cast<uint8_t>(l >> 16), l1, l0};
+}
+
+inline bytes wrap_list(const bytes& content)
+{
+    return internal::encode_length<0xc0, 0xf7>(content.size()) + content;
```

Comment:
perhaps use decimals because YP defines them as decimals, and also `short_cutoff` on line 24
```suggestion
    return internal::encode_length<192, 247>(content.size()) + content;
```